src/org/zaproxy/zap/extension/fuzz/payloads/ui/impl/JsonPayloadGeneratorAdapterUIHandler.java

```diff
@@ -0,0 +1,233 @@
+        }
+
+        @Override
+        public JsonPayloadGenerator getPayloadGenerator() {
+            return jsonPayloadGenerator;
+        }
+
+        @Override
+        public PayloadGeneratorUI<DefaultPayload, JsonPayloadGenerator> copy() {
+            return this;
+        }
+    }
+
+    public static class JsonPayloadGeneratorUIPanel extends AbstractPersistentPayloadGeneratorUIPanel<DefaultPayload, JsonPayloadGenerator, JsonPayloadGeneratorUI> {
+        private static final String JSON_FIELD_LABEL = getString("original.field.label");
+        private static final String NUMBER_PAYLOADS_LABEL = getString("number.payloads.label");
+
+        private JPanel fieldsPanel;
+        private ZapTextArea jsonTextArea;
+        private ZapNumberSpinner numberOfPayloadsSpinner;
+
+        private JsonPayloadGeneratorUI oldGenerator;
+
+        public JsonPayloadGeneratorUIPanel() {
+            fieldsPanel = new JPanel();
+
+            GroupLayout layout = new GroupLayout(fieldsPanel);
+            fieldsPanel.setLayout(layout);
+            layout.setAutoCreateGaps(true);
+
+            numberOfPayloadsSpinner = new ZapNumberSpinner(1, 1, Integer.MAX_VALUE);
+            JLabel numberOfPayloadsLabel = new JLabel(NUMBER_PAYLOADS_LABEL);
+            numberOfPayloadsLabel.setLabelFor(numberOfPayloadsSpinner);
+
+            JLabel jsonFieldLabel = new JLabel(JSON_FIELD_LABEL);
+            jsonFieldLabel.setLabelFor(getJsonTextArea());
+            JScrollPane jsonFieldScrollPane = new JScrollPane(getJsonTextArea());
+
+            layout.setHorizontalGroup(layout.createSequentialGroup()
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.TRAILING)
+                                    .addComponent(numberOfPayloadsLabel)
+                                    .addComponent(jsonFieldLabel))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.LEADING)
+                                    .addComponent(numberOfPayloadsSpinner)
+                                    .addComponent(jsonFieldScrollPane)));
+
+            layout.setVerticalGroup(layout.createSequentialGroup()
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(numberOfPayloadsLabel)
+                                    .addComponent(numberOfPayloadsSpinner))
+                    .addGroup(
+                            layout.createParallelGroup(GroupLayout.Alignment.BASELINE)
+                                    .addComponent(jsonFieldLabel)
+                                    .addComponent(jsonFieldScrollPane)));
+        }
+
+        @Override
+        protected JsonPayloadGenerator getPayloadGenerator() {
+            if (!validate()) {
+                return null;
+            }
+            String j = jsonTextArea.getText();
```

Comment:
This could be inlined (e.g. `new JsonPayloadGenerator(jsonTextArea.getText(), numberOfPayloadsSpinner.getValue())`).